testcases/misc_tests/obj_mgmt.c

```diff
@@ -1162,251 +1162,260 @@ CK_RV do_CreateTokenObjects(void)
+		{CKA_KEY_TYPE,  &obj2_type,     sizeof(obj2_type)},
+		{CKA_LABEL,     obj2_label,     sizeof(obj2_label)-1},
+		{CKA_VALUE,     obj2_data,      sizeof(obj2_data)}
 	};
 
-        if (skip_token_obj == TRUE) {
-                testcase_notice("Skipping tests that creates token objects");
-                return CKR_OK;
+    CK_OBJECT_HANDLE h_counter1,
+                     h_clock,
+                     h_obj1,
+                     h_obj2,
+                     obj_list[10];
+
+    CK_ATTRIBUTE find_tmpl[] = {
+        {CKA_CLASS,	&counter1_class, sizeof(counter1_class)}
+    };
+
+    if (skip_token_obj == TRUE) {
+        testcase_notice("Skipping tests that creates token objects");
+        return CKR_OK;
+    }
+
+    slot_id = SLOT_ID;
+
+    testcase_begin("starting...");
+
+    if (get_user_pin(user_pin))
+        return CKR_FUNCTION_FAILED;
+
+    user_pin_len = (CK_ULONG)strlen((char *)user_pin);
+
+    /* Open a session with the token */
+    rc = funcs->C_OpenSession(slot_id,
+                              (CKF_SERIAL_SESSION|CKF_RW_SESSION),
+                              NULL_PTR,
+                              NULL_PTR,
+                              &h_session);
+    if (rc != CKR_OK) {
+        testcase_fail("C_OpenSession() rc = %s", p11_get_ckr(rc));
+        goto done;
+    }
+
+    // Login correctly
+    rc = funcs->C_Login(h_session, CKU_USER, user_pin, user_pin_len);
+    if (rc != CKR_OK) {
+        testcase_fail("C_Login() rc = %s", p11_get_ckr(rc));
+        goto session_close;
+    }
+
+    /* Create the 4 test objects */
+    rc = funcs->C_CreateObject(h_session, obj1_template, 4, &h_obj1);
+    if (rc != CKR_OK) {
+        testcase_fail("C_CreateObject() rc = %s", p11_get_ckr(rc));
+		return rc;
+	}
+
+    rc = funcs->C_CreateObject(h_session, obj2_template, 5, &h_obj2);
+    if (rc != CKR_OK) {
+        testcase_fail("C_CreateObject() rc = %s", p11_get_ckr(rc));
+        goto destroy_1;
+    }
+
+    /* try and create a monotonic object. This should fail
+     * since it is a read only feature.
+     */
+    rc = funcs->C_CreateObject(h_session, counter1_template, 6, &h_counter1);
+    if (rc != CKR_ATTRIBUTE_READ_ONLY) {
+        testcase_fail("C_CreateObject() rc = %s", p11_get_ckr(rc));
+        goto destroy_2;
+    }
+
+    rc = funcs->C_CreateObject(h_session, clock_template, 4, &h_clock);
+    if (rc != CKR_OK) {
+        testcase_fail("C_CreateObject() rc = %s", p11_get_ckr(rc));
+        goto destroy_2;
+    }
+
+
+    /* Search for the 2 objects w/o HW_FEATURE set */
+    /* A NULL template here should return all objects in v2.01, but
+     * in v2.11, it should return all objects *except* HW_FEATURE
+     * objects.
+     */
+    rc = funcs->C_FindObjectsInit(h_session, NULL, 0);
+    if (rc != CKR_OK) {
+        testcase_fail("C_FindObjectsInit() rc = %s", p11_get_ckr(rc));
+        goto destroy;
+    }
+
+    rc = funcs->C_FindObjects(h_session, obj_list, 10, &find_count);
```

Comment:
Similar question here, why 10?

Reply:
The third parameter in a C_FindObjects refers to the maximum number of objects handles to be returned, in this case being the size of the obj_list array.
For more information on this, please check page 96 of pkcs11-base-v2.40.pdf.